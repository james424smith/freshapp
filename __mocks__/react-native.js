import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
