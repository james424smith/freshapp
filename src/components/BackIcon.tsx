import React from "react";
import { TouchableOpacity } from "react-native";

import Back from "../../assets/icons/bottom_navigation/backArrow.svg";
import { Theme, useNavigation, useTheme } from "@react-navigation/native";

interface Props {
  goBack: boolean;
  path?: string;
  fill?: string;
}

const size = 25;

export default (props: Props) => {
  const { goBack, path, fill } = props;
  const navigation = useNavigation();
  const theme: Theme = useTheme();
  const handleOnPress = (goBackLocal: boolean, pathLocal?: string) => {
    if (!goBackLocal && pathLocal) {
      navigation.navigate(pathLocal);
    } else {
      navigation.goBack();
    }
  };

  return (
    <TouchableOpacity
      testID={"back-button"}
      hitSlop={{ top: 5, bottom: 5, left: 5, right: 5 }}
      onPress={() => handleOnPress(goBack, path)}
    >
      <Back
        fill={fill ?? theme.colors.blackAndWhite}
        width={size}
        height={size}
      />
    </TouchableOpacity>
  );
};
