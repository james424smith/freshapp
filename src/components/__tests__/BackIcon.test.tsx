import React from "react";
import BackIcon from "../BackIcon";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import { useNavigation } from "@react-navigation/native";

describe("BackIcon snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui and press icon when goBack is true", () => {
    const { toJSON, getByTestId } = render(<BackIcon goBack={true} />);

    fireEvent.press(getByTestId("back-button"));
    expect(useNavigation().goBack).toHaveBeenCalledTimes(1);
    expect(useNavigation().navigate).toHaveBeenCalledTimes(0);
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render correct ui and press icon when goBack is false", () => {
    const { toJSON, getByTestId } = render(
      <BackIcon goBack={false} path="Assignment" />
    );

    fireEvent.press(getByTestId("back-button"));
    expect(useNavigation().goBack).toHaveBeenCalledTimes(0);
    expect(useNavigation().navigate).toHaveBeenCalledWith("Assignment");
    expect(toJSON()).toMatchSnapshot();

    expect(toJSON()).toMatchSnapshot();
  });
});
