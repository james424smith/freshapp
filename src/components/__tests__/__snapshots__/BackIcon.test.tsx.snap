// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon snapshot test should render correct ui and press icon when goBack is false 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="back-button"
>
  <<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><defs><style>.a{fill:none;}.b{fill-rule:evenodd;}</style></defs><rect class="a" width="24" height="24"/><path class="b" d="M15.4,7.4,14,6,8,12l6,6,1.4-1.4L10.8,12Z"/></svg>
    height={25}
    width={25}
  />
</View>
`;

exports[`BackIcon snapshot test should render correct ui and press icon when goBack is false 2`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="back-button"
>
  <<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><defs><style>.a{fill:none;}.b{fill-rule:evenodd;}</style></defs><rect class="a" width="24" height="24"/><path class="b" d="M15.4,7.4,14,6,8,12l6,6,1.4-1.4L10.8,12Z"/></svg>
    height={25}
    width={25}
  />
</View>
`;

exports[`BackIcon snapshot test should render correct ui and press icon when goBack is true 1`] = `
<View
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="back-button"
>
  <<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><defs><style>.a{fill:none;}.b{fill-rule:evenodd;}</style></defs><rect class="a" width="24" height="24"/><path class="b" d="M15.4,7.4,14,6,8,12l6,6,1.4-1.4L10.8,12Z"/></svg>
    height={25}
    width={25}
  />
</View>
`;
